### ISSUE TITLE
Package request: gVisor

### ISSUE THREAD

**Issue by lars18th** [termux/termux-packages#4901]
**Package description**
This package provides a non-root (user land) option to execute containers (aka Docker).

**Link to home page and sources**
1. Home page: https://gvisor.dev/
2. Source code: https://github.com/google/gvisor

**Additional information**
* The current implementation includes _some_ ARM support (WIP).
* The project provides the binary `runsc` that runs a container without root privileges.
* With this tool it will be viable in the future to run some kind of "termux-on-termux", with the objective of virtualizing some parts of Linux (filesystems, networking, etc) and then execute a new Termux instance inside a container without the restrictions imposed by the Android environment. An example: "mount" a loopback image from the SD-card (this will be possible inside the virtualized container).

I hope someone wants to work in porting this package to Termux.
Regards.

**Comment by ifurther** [comment]
It should need docker? `/bin/sh: 2: docker: not found`

**Comment by xeffyr** [comment]
Yes, but it can use RunC as well which is available in root repo.

However package description is not true:
> This package provides a non-root (user land) option to execute containers (aka Docker).

Rootless containerization possible only with custom kernels where user namespaces are enabled.

Package is kinda useless for the most users.

**Comment by xeffyr** [comment]
Project dependencies: https://github.com/google/gvisor#requirements

**Comment by lars18th** [comment]
Hi @xeffyr ,

> Those who want to try "root (less?) containerization", can use runc package from root repo.

Ok, then users without root don't have any alternative at time.

Thank you very much, though!

**Comment by azazar** [comment]
> It should need docker? `/bin/sh: 2: docker: not found`

No. It doesn't need docker. Docker could use it if it's available. Also podman/udocker can support it.

**Comment by azazar** [comment]
> Rootless containerization possible only with custom kernels where user namespaces are enabled.

It seems that [gvisor doesn't require host kernel to support namespaces when it's using EROFS filesystem](https://groups.google.com/g/gvisor-users/c/DyIXSbxNOKI).

**Comment by robertkirkman** [comment]
Update on this topic including code has been posted in the other issue
